### ISSUE TITLE
Missing convert method for `OneOfAPIModel`

### ISSUE THREAD

**Issue by avandecreme** [JuliaComputing/OpenAPI.jl#32]
Hello,

First thanks for writing and publishing that library.
I am trying it out on my company's API.
Here is a first issue I had.

<details>
<summary>Given the following schema</summary>

```json
{
    "components": {
        "schemas": {
            "Beer": {
                "enum": [
                    "IPA",
                    "Lager"
                ],
                "type": "string"
            },
            "Drink": {
                "oneOf": [
                    {
                        "$ref": "#/components/schemas/Beer"
                    },
                    {
                        "$ref": "#/components/schemas/Wine"
                    }
                ]
            },
            "Wine": {
                "enum": [
                    "RedWine",
                    "WhiteWine"
                ],
                "type": "string"
            }
        }
    },
    "info": {
        "title": "",
        "version": ""
    },
    "openapi": "3.0.0",
    "paths": {
        "/bar": {
            "get": {
                "responses": {
                    "200": {
                        "content": {
                            "application/json;charset=utf-8": {
                                "schema": {
                                    "$ref": "#/components/schemas/Drink"
                                }
                            }
                        },
                        "description": ""
                    }
                }
            }
        }
    }
}
```
</details>

And running the following test code (where `julia` is the directory where the client code was generated by openapi-generator):
```julia
include("julia/src/APIClient.jl")

import Base: convert
import OpenAPI
import OpenAPI.Clients: Client

client = Client("http://localhost:8081")

api = APIClient.DefaultApi(client)

# convert(::Type{T}, v) where {T<:OpenAPI.OneOfAPIModel} = T(v)

drink, _ = APIClient.bar_get(api)

println(drink)
```

I get: `ERROR: LoadError: MethodError: Cannot 'convert' an object of type String to an object of type Main.APIClient.Drink`

Uncommenting the `convert(::Type{T}, v) where {T<:OpenAPI.OneOfAPIModel} = T(v)` fixes the issue.

Should that line be added to `client.jl`? Note that I am entirely new to Julia, so that might well be stupid.

**Comment by avandecreme** [comment]
In the end, I had to add all those methods:
```julia
convert(::Type{T}, v::T) where {T<:OpenAPI.OneOfAPIModel} = v
convert(::Type{T}, json::Dict{String,Any}) where {T<:OpenAPI.OneOfAPIModel} = OpenAPI.from_json(T, json)
convert(::Type{T}, v) where {T<:OpenAPI.OneOfAPIModel} = T(v)
convert(::Type{T}, v::T) where {T<:OpenAPI.AnyOfAPIModel} = v
convert(::Type{T}, json::Dict{String,Any}) where {T<:OpenAPI.AnyOfAPIModel} = OpenAPI.from_json(T, json)
convert(::Type{T}, v) where {T<:OpenAPI.AnyOfAPIModel} = T(v)
```

**Comment by tanmaykm** [comment]
fixed by #34
